cdap-kubernetes/src/main/java/io/cdap/cdap/k8s/runtime/KubeTwillLauncher.java

```diff
@@ -157,4 +166,70 @@ private void deletePod(PodInfo podInfo) {
       LOG.warn("Failed to delete pod {} with uid {}", podInfo.getName(), podInfo.getUid(), e);
     }
   }
+
+  /**
+   * Get instance id from the annotation. The annotation will have array of boolean
+   */
+  private InstanceInfo getInstanceInfo(PodInfo podInfo) throws Exception {
+    int instanceId = 0;
+    int instanceCount = 1;
+
+    if (podInfo.getOwnerReferences().stream().anyMatch(r -> "Job".equals(r.getKind()))) {
+      String jobName = podInfo.getOwnerReferences().get(0).getName();
+      ApiClient client = Config.defaultClient();
+      BatchV1Api batchApi = new BatchV1Api(client);
+      Random rand = new Random();
+
+      try {
+        boolean done = false;
+        int randomNumber = 0;
+
+        while (!done) {
+          V1Job v1Job = batchApi.readNamespacedJob(jobName, podInfo.getNamespace(), null, null, null);
+          Map<String, String> annotations = v1Job.getMetadata().getAnnotations();
+          if (!annotations.containsKey(JOB_INSTANCES)) {
```

Comment:
since this doesn't change for each iteration through the loop, would be better to do this check before the loop begins to avoid checking each loop and to make the loop contents shorter and more readable.